src/main/java/interfaces/IServerSocket.java

```diff
@@ -0,0 +1,8 @@
+package interfaces;
+
+import java.util.List;
+
+public interface IServerSocket {
```

Comment:
**suggestion**: Name your interfaces for the role they represent, instead of the thing you want to implement.

What role or capability will this abstraction serve? Maybe we could name it along those lines?